pkg/engine/shield/manager.go

```diff
@@ -41,3 +42,15 @@ func (mgr *Manager) IsShielded(target key.TargetID) bool {
 	}
 	return false
 }
+
+func (mgr *Manager) MaxShield(target key.TargetID) int {
+	maxShield := mgr.targets[target][0].hp
+	index := 0
+	for i, shield := range mgr.targets[target] {
+		if shield.hp > maxShield {
+			maxShield = shield.hp
+			index = i
+		}
+	}
+	return index
+}
```

Comment:
* The current implementation will cause the sim to crash (and does for our integration tests). Consider the case when the target has no shields (length of `activeShields` is 0 so getting a shield at index 0 is impossible)
* This function is named with a capital letter (which makes it public). I think that makes sense, but the return does not make sense for a public function. If I am some other service and I call `MaxShield`, returning the index tells me nothing since I have no way to access shields by index (and probably shouldn't). Given how `MaxShield` is currently used, I think it makes more sense for this to just return the `hp` of the max shield (or `0` if the target has no shields)